kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/policy/KeepLatestContainerOnlyPolicy.java

```diff
@@ -0,0 +1,131 @@
+ *     <li>policy.klo.unit - time unit that the interval was specified in - if not given milliseconds are assumed</li>
+ * </ul>
+ *
+ * Name of this policy (to be used to activate it) is <code>KeepLatestOnly</code>
+ */
+public class KeepLatestContainerOnlyPolicy implements Policy {
+
+    private static final Logger logger = LoggerFactory.getLogger(KeepLatestContainerOnlyPolicy.class);
+
+    private static final String INTERVAL_VALUE = "policy.klo.interval";
+    private static final String INTERVAL_TIME_UNIT = "policy.klo.unit";
+
+    private static final String NAME = "KeepLatestOnly";
+
+    private long interval;
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public long getInterval() {
+        return interval;
+    }
+
+    @Override
+    public void start() {
+        logger.debug("Starting KeepLatestContainerOnlyPolicy policy...");
+        TimeUnit timeUnit = TimeUnit.valueOf(System.getProperty(INTERVAL_TIME_UNIT, TimeUnit.MILLISECONDS.toString()));
+        long givenInterval = Long.parseLong(System.getProperty(INTERVAL_VALUE, "0"));
+
+        if (givenInterval > 0) {
+            timeUnit.convert(givenInterval, timeUnit);
+            this.interval = TimeUnit.MILLISECONDS.convert(givenInterval, timeUnit);
+        } else {
+            // default once a day
+            this.interval =  TimeUnit.MILLISECONDS.convert(1L, TimeUnit.DAYS);
+        }
+        logger.debug("Started {} policy", this);
+    }
+
+    @Override
+    public void stop() {
+        this.interval = -1;
+        logger.debug("Stopped {} policy", this);
+    }
+
+    @Override
+    public void apply(KieServerRegistry kieServerRegistry, KieServer kieServer) {
+
+        DefaultRestControllerImpl controller = new DefaultRestControllerImpl(kieServerRegistry);
+
+        List<String> containerAliases = kieServerRegistry.getContainerAliases();
+        if (containerAliases.isEmpty()) {
+            logger.debug("No containers found, quiting");
+            return;
+        }
+
+        for (String alias : containerAliases) {
+
+            List<KieContainerInstanceImpl> containerInstances = kieServerRegistry.getContainersForAlias(alias);
+            if (containerInstances.isEmpty() || containerInstances.size() == 1) {
+                logger.debug("Containers for alias {} are already on expected level (number of containers is {})", alias, containerInstances.size());
+                continue;
+            }
+
+            String latestContainerId = LatestContainerLocator.get().locateContainer(alias, containerInstances);
+            final Map<String, String> report = new HashMap<>();
+            containerInstances.stream()
+                    .filter(kci -> !kci.getContainerId().equals(latestContainerId))
+                    .forEach(kci -> {
+                        ServiceResponse<Void> response = kieServer.disposeContainer(kci.getContainerId());
```

Comment:
No, you are correct. We have to make the dispose() call safe (in case it is not) and not put the burden on the policy. +1